codex-rs/core/src/codex.rs

```diff
@@ -1474,6 +1478,101 @@ async fn submission_loop(
                 };
                 sess.send_event(event).await;
             }
+            Op::RunUserShellCommand { command } => {
```

Comment:
This `submission_loop()` function needs to be refactored, so rather than pile on, can we move the handler for `Op::RunUserShellCommand` into its own function?